### PR TITLE
murdock: chunked get_jobs()

### PR DESCRIPTION
**Pull request by kaspar030**
### Contribution description

Previously, murdock would first collect all jobs, then start building. This PR collects a couple of changes to make job creation happen in parallel to job collection.

The effect is that the first build jobs will start before job collection is complete. You'll notice that as soon as the percentage bar appears, the first jobs (1000 - 5000, depending on the performance of the workers that process job collection) complete really fast, as they're actually already done and only the result gets processed. This should cut the total build time by at least 30 seconds, and mitigate the impact of a slow worker taking it's time to process it's part of "get_jobs()".

### Testing procedure

- confirm all jobs are still built
- maybe re-trigger this PR while watching the output. Concentrate on what happens after "collecting jobs..." is done and the percentage bar appears.

### CODE REVIEW

**Comment by miri64** [comment]
Shouldn't the jobs number increase then in the frontend, while the jobs are building? Or am I completely misunderstanding this PR?

File: .murdock [hunk 1]
```diff
@@ -134,8 +145,8 @@ get_app_board_toolchain_pairs() {
 get_compile_jobs() {
     get_apps | \
         dwqc ${DWQ_ENV} -s \
-        "$0 get_app_board_toolchain_pairs \${1}" \
-        | xargs '-d\n' -n 1 echo $0 compile
+        ${DWQ_JOBID:+--subjob} \
+        "$0 get_app_board_toolchain_pairs \${1} ./.murdock compile"
```
Comment (miri64) [inline .murdock]:
```suggestion
        "$0 get_app_board_toolchain_pairs \${1} $0 compile"
```

File: .murdock [hunk 2]
```diff
@@ -131,7 +136,7 @@ get_app_board_toolchain_pairs() {
     do
         for toolchain in $(get_supported_toolchains $appdir $board)
         do
-            echo $appdir $board:$toolchain
+            echo $prefix $appdir $board:$toolchain
```
Comment (miri64) [inline .murdock]: Ok, now I understand. Adding this...

File: .murdock [hunk 3]
```diff
@@ -140,8 +145,7 @@ get_app_board_toolchain_pairs() {
 get_compile_jobs() {
     get_apps | \
         dwqc ${DWQ_ENV} -s \
-        "$0 get_app_board_toolchain_pairs \${1}" \
-        | xargs '-d\n' -n 1 echo $0 compile
```
Comment (miri64) [inline .murdock]: ... makes the piping to `xargs` here this unnecessary.

File: .murdock [hunk 4]
```diff
@@ -125,7 +125,7 @@ get_app_board_toolchain_pairs() {
 
     # collect extra arguments into prefix variable
     shift
-    local prefix="$@"
+    local prefix="$*"
```
Comment (miri64) [inline .murdock]: I believe the `$@` => `$*` change accidentally was added to 1818da495262fb9ed6d39dc5b775bd25d116d6f0.

File: .murdock (hunk 1 above)
Comment (kaspar030) [reply .murdock]: yup, fixed

File: .murdock (hunk 3 above)
Comment (kaspar030) [reply .murdock]: yup. the commit message of the first commit also has some explanation.

File: .murdock (hunk 4 above)
Comment (kaspar030) [reply .murdock]: the change is necessary. it belongs to the first commit, though... I'll fix that.

Comment (kaspar030) [reply .murdock]: done

**Comment by kaspar030** [comment]
I'll squash this. @miri64 it includes your suggested fix and the move of "$@" -> "$*" into the first commit.

**Comment by kaspar030** [comment]
Thanks for reviewing!

**Pull request merged**